File: src/main/java/org/reflections/scanners/TypeAnnotationsScanner.java
package org.reflections.scanners;

import java.lang.annotation.Inherited;
import java.util.List;

/**
 * scans for class's annotations, where @Retention(RetentionPolicy.RUNTIME)
 */
@SuppressWarnings("unchecked")
public class TypeAnnotationsScanner extends AbstractScanner {

    @Override
    public void scan(Object cls) {
        String className = getMetadataAdapter().getClassName(cls);

        for (String annotationType : (List<String>) getMetadataAdapter().getClassAnnotationNames(cls)) {

            if (acceptResult(annotationType)
                || annotationType.equals(Inherited.class.getName())) { //as an exception, accept Inherited as well
                getStore().put(annotationType, className);
            }
        }
    }

}

File: src/main/java/org/reflections/scanners/TypeAnnotationsScanner.kt
package org.reflections.scanners

import org.reflections.adapters.ClassAdapter
import java.lang.annotation.Inherited

/**
 * scans for class's annotations, where @Retention(RetentionPolicy.RUNTIME)
 */
class TypeAnnotationsScanner : AbstractScanner() {

    override fun scan(cls: ClassAdapter) {
        val className = cls.name

        for (annotationType in cls.annotations) {

            if (acceptResult(annotationType) || annotationType == Inherited::class.java.name) { //as an exception, accept Inherited as well
                store!!.put(annotationType, className)
            }
        }
    }

}
